TF-288 Required Encryption Password (#1583)
hashicorp/terraform-website: content/source/docs/enterprise/install/encryption-password.html.md

# Terraform Enterprise Encryption Password

> When using internally-managed Vault, Terraform Enterprise requires that the operator specify a password that will be

Reviewer: ```suggestion
When using internally managed Vault, Terraform Enterprise requires that the operator specify a password that will be
```

Minor suggestion - I don't think we need this to be hyphenated, but I'm open to being wrong.

Author: I personally don't have a preference. I started this pull request without the hyphen and changed to the hyphen from another review.

Reviewer: Thinking about it a bit more, I think there's some precedent for omitting the hyphen. Specifically, we don't use "externally-managed" when referring to [external Vault](https://www.terraform.io/docs/enterprise/before-installing/vault.html). The only other place I could find that we use the phrase "internally managed" when referring to Vault is in the [VMWare reference architecture](https://www.terraform.io/docs/enterprise/before-installing/reference-architecture/vmware.html#infrastructure-requirements) and it doesn't use the hyphen.

Still not wildly opinionated on it, but was trying to make sure we were consistent.

Reviewer 2: Speaking as the original reviewer- lets be consistent, the hyphen was a pedantic (and opinionated) grammatical switch. Apologies for the noise, I should have looked at the docs as a whole before asking for a pedantic change.

> used to to encrypt and decrypt the internally-managed Vault unseal key and root token. This password is called the

Reviewer: ```suggestion
used to to encrypt and decrypt the internally managed Vault unseal key and root token. This password is called the
```

Feel free to dismiss this if you disagree re: hyphen

> "encryption password". Please be sure to retain this value as it will be needed in the event of a re-installation.

Reviewer: Were these line breaks intentional?

Author: Yep! Just folding lines at the 120 column instead of the 80 column like before.

> The encryption password is used to protect the internally-managed Vault unseal key and root token with a password

Reviewer: ```suggestion
The encryption password is used to protect the internally managed Vault unseal key and root token with a password
```

Hyphen again

> provided by the operator. It allows Terraform Enterprise to securely store the Vault unseal key and root token in

Reviewer: Same question here

Author: Yep! Just folding lines at the 120 column instead of the 80 column like before.

# Terraform Enterprise Encryption Password

> When using internally managed Vault, Terraform Enterprise requires that the operator specify a password that will be

Reviewer 2: ```suggestion
When using internally-managed Vault, Terraform Enterprise requires that the operator specify a token value that will be
```

> used to to encrypt and decrypt the internally managed Vault unseal key and root token. This pasword is called the

Reviewer 2: ```suggestion
used to encrypt and decrypt the internally-managed Vault unseal key and root token. This password is called the
```

> encryption password. Please be sure to retain the value of the encryption password as it will be needed in the of a

Reviewer 2: ```suggestion
"encryption password". Please be sure to retain this value as it will be needed in event the of a
```

re-installation.

> The encryption password is used to protect the internally managed Vault unseal key and root token with a password

Reviewer 2: ```suggestion
The encryption password is used to protect the internally-managed Vault unseal key and root token with a token
```

> provided by the operator. It allows us to store the Vault unseal key and root token in PostgreSQL, which means that

Reviewer 2: ```suggestion
provided by the operator. It allows TFE to securely store the Vault unseal key and root token in PostgreSQL, which means that
```
